### PR TITLE
Use specific type declaration for "auditors" variable

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##
This PR makes the type declaration more specific for the `auditors` variable.

As long as I was cleaning things up, I also took this opportunity to improve the `auditors` variable description by listing the currently-defined audit roles in the description instead of in a code comment.

## 💭 Motivation and context ##
In https://github.com/cisagov/cool-auditor-iam/pull/1#pullrequestreview-584542093, @jsf9k mentioned (after #1 had been merged) that

> `map` is shorthand for `map(any)` and should be eschewed.

## 🧪 Testing ##
I ran a successful `terraform apply` and verified that no changes were needed, as expected.  

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
* [x] All new and existing tests pass.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -5,9 +5,8 @@
 # ------------------------------------------------------------------------------
 
 variable "auditors" {
-  type = map
-  # Currently-defined audit roles: financial_audit, security_audit
-  description = "A map containing the usernames of each auditor and a list of audit roles assigned to that user.  Example: { \"firstname1.lastname1\" = { \"roles\" = [ \"financial_audit\", \"security_audit\" ] },  \"firstname2.lastname2\" = { \"roles\" = [ \"security_audit\" ] } }"
+  type        = map(map(list(string)))
```
Comment (jsf9k) [inline variables.tf:8]: :fire:

**Review by jsf9k — APPROVED** [review]
Thank you for circling back to this.

File: variables.tf (hunk 1 above)
Comment (mcdonnnj) [reply variables.tf:8]: 🔥

**Review by mcdonnnj — APPROVED** [review]
It's gnarly, but I'm glad to see it going in here as well.

**Pull request merged**